src/NUnitFramework/framework/Constraints/EmptyConstraint.cs

```diff
@@ -48,11 +50,13 @@ public override string Description
         /// <returns>True for success, false for failure</returns>
         public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
-            // NOTE: actual is string will fail for a null typed as string
+            // NOTE: actual is string will fail for a null typed as string           
+            Type actualType = typeof(TActual);
```

Comment:
If you have the variable you should use it everywhere tyepof(TActual) is used i.e. line 56